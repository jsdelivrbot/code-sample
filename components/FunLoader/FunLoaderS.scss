@import "guide/includes";

.FunLoader {
  position: relative;
  padding: 10px;

  .fa-grav {
    font-size: 40px;
    animation: spin 1.5s infinite;
  }

  @keyframes spin {
    from {
      transform:rotate(0deg);
      color: $primary-color;
    }
    to {
      transform:rotate(360deg);
      color: $secondary-color;
    }
  }

  &::before,
  &::after {
    z-index: 1;
    box-sizing: border-box;
    content: '';
    position: absolute;
    border: 3px solid transparent;
    width: 0;
    height: 0;
    animation-timing-function: linear;
  }

  &::before {
    top: 0;
    left: 0;
    border-top-color: $primary-color;
    border-right-color: $primary-color;
    animation: border-before 1.5s infinite;
    animation-direction: alternate;
  }

  &::after {
    bottom: 0;
    right: 0;
    border-bottom-color: $secondary-color;
    border-left-color: $secondary-color;
    animation: border-after 1.5s infinite;
    animation-direction: alternate;
  }

  @keyframes border-before {
    0% {
      width: 0;
      height: 0;
      border-right-color: transparent;
    }
    24.99% {
      border-right-color: transparent;
    }
    25% {
      height: 0;
      width: 100%;
      border-right-color: $primary-color;
    }
    50%,
    100% {
      width: 100%;
      height: 100%;
    }
  }

  @keyframes border-after {
    0%,
    49.99% {
      width: 0;
      height: 0;
      border-left-color: transparent;
      border-bottom-color: transparent;
    }
    50% {
      border-bottom-color: $secondary-color;
    }
    74.99% {
      border-left-color: transparent;
    }
    75% {
      height: 0;
      width: 100%;
      border-left-color: $secondary-color;
      border-bottom-color: $secondary-color
    }
    100% {
      width: 100%;
      height: 100%;
    }
  }
}
