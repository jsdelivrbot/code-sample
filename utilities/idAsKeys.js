import keyBy from 'lodash/keyBy'
import { array, checkPropTypes } from 'prop-types'

const propTypes = {
  items: array,
}

const idAsKeys = function ({ items }) {
  checkPropTypes(propTypes, arguments[0], 'prop', 'idAsKeys')

  return keyBy(items, (item) => {
    return item.data.id
  })
}

export default idAsKeys
