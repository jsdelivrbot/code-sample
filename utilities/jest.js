export const describe = global.describe

export const it = global.it

export const expect = global.expect

export const beforeEach = global.beforeEach

export const afterEach = global.afterEach
