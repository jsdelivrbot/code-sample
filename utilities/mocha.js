export const describe = window.describe

export const it = window.it

export default window.mocha
