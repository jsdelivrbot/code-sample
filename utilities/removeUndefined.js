import merge from 'lodash/merge'
import forIn from 'lodash/forIn'
import isObject from 'lodash/isObject'

// TODO: use destructor
function removeUndefined (o) {
  if (isObject(o) && !Array.isArray(o)) {
    let obj = merge({}, o)

    forIn(obj, function (val, key) {
      if (val === null || val === undefined) {
        delete obj[key]
      } else {
        obj[key] = removeUndefined(val)
      }
    })

    return obj
  }

  return o
}

export default removeUndefined
