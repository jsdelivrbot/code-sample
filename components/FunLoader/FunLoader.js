import './FunLoaderS.scss'

import React from 'react'
import classNames from 'classnames'

const FunLoader = ({ small = false, smallest = false }) => {
  const classes = {
    FunLoader: true,
    small,
    smallest,
  }

  return (
    <div className={classNames(classes)}>
      <span className='fa fa-grav' />
    </div>
  )
}

FunLoader.propTypes = {
  small: React.PropTypes.bool,
  smallest: React.PropTypes.bool,
}

export default FunLoader
