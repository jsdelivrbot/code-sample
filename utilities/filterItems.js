import filter from 'lodash/filter'
import isMatch from 'lodash/isMatch'
import removeUndefined from 'utilities/removeUndefined'
import { object, checkPropTypes } from 'prop-types'

const propTypes = {
  items: object,
  filters: object,
}

const filterItems = function ({ items, filters }) {
  checkPropTypes(propTypes, arguments[0], 'prop', 'filterItems')

  const removedUndefinedFilters = removeUndefined(filters)

  return filter(items, ({ data }) => {
    return isMatch(data, removedUndefinedFilters)
  })
}

export default filterItems
