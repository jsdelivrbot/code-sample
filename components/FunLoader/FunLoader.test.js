import FunLoader from './FunLoader'
import React from 'react'
import renderer from 'react-test-renderer'
import { describe, it, expect } from 'utilities/jest'

describe('Test FunLoader', () => {
  it('should be able to render the component', () => {
    const component = renderer.create(
      <FunLoader />
    )

    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
