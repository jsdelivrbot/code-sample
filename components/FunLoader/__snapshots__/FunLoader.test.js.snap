// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FunLoader should be able to render the component 1`] = `
<div
  className="FunLoader"
>
  <span
    className="fa fa-grav"
  />
</div>
`;
